Improving documentation of `models.encodings.entangling_layer` (#1273)
qiboteam/qibo: doc/source/api-reference/qibo.rst

""""""""""""""""

Generates a layer of nearest-neighbour two-qubit gates, assuming 1-dimensional connectivity.
With the exception of :class:`qibo.gates.gates.GeneralizedfSim`,

> can be set to any of the two-qubit gates implemented in ``qibo``.

Reviewer: ```suggestion
any of the two-qubit gates implemented in ``qibo`` can be selected to customize the entangling layer.
```

If the chosen gate is parametrized, all phases are set to :math:`0.0`.
Note that these phases can be updated a posterior by using
:meth:`qibo.models.Circuit.set_parameters`.

> There are four possible choices of layer ``architechture``:

Reviewer: ```suggestion
There are four possible choices of layer ``architecture``:
```

``diagonal``, ``shifted``, ``even-layer``, and ``odd-layer``.

> For instance, we show below an example for each architecture and ``nqubits = 6``.

Author: ```suggestion
For instance, we show below an example of each architecture for ``nqubits = 6``.
```
